process-manager.js

```diff
@@ -43,18 +43,14 @@ class ProcessWrapper extends EventEmitter {
 }
 
 class ProcessManager {
-	constructor(options) {
-		if (!options) options = {};
-
+	constructor(execFile, maxProcesses, isChatBased) {
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = '' + execFile;
+		this.maxProcesses = (!isNaN(maxProcesses) || maxProcesses >= 1) ? maxProcesses : 1;
+		this.isChatBased = !!isChatBased;
 
```

Comment:
Why not read each of these individually from 'options'?

There is no good reason why 'isChatBased' should have a whole argument for it.